assignment2/Usermanager.java

```diff
@@ -13,112 +13,114 @@
-        //handling other options
-        if (order < 1 || order > 2 || ch < 1 || ch > 4) 
-        {
-            System.out.println("Invalid Option ! ");
-            in.close();
-            return;
-        }    
-        System.out.println("------------------------------------------------------------------------------------------------------------");
-        System.out.println("Name\t\tRoll Number\t\tAge\t\tAddress\t\tCourses");
-        System.out.println("---------------------------------------------------------------------------------------------------------");
-        user.stream().sorted((a, b) -> {
-            int result = 0;
-            switch (ch) {
-                case 1:
-                    result = a.getName().compareTo(b.getName());
-                    break;
-                case 2:
-                    result = a.getRollNumber() - b.getRollNumber();
-                    break;
-                case 3:
-                    result = a.getAge() - b.getAge();
-                    break;
-                case 4:
-                    result = a.getAddress().compareTo(b.getAddress());
-                    break;
-            }
-         //If descending order is selected , invert the sign of result
-            if (order == 2) result = -result;
-            return result;
-        }).forEach(u -> u.display());
-        in.close();		
+		System.out.println("2. Roll number");
+		System.out.println("3. Age");
+		System.out.println("4. Address");
+		System.out.print("Fields to be choosed for sorting (Enter choice) :  ");        
+		ch = Integer.parseInt(in.nextLine());
+		System.out.println("1. Ascending order ");
+		System.out.println("2. Descending order");
+		System.out.print("Enter choice : ");
+		order = Integer.parseInt(in.nextLine());
+		//handling for other options
+		if (order < 1 || order > 2 || ch < 1 || ch > 4) 
+		{
+			System.out.println("Invalid Option ! ");
+			in.close();
+			return;
+		}    
+		System.out.println("------------------------------------------------------------------------------------------------------------");
+		System.out.println("Name\t\tRoll Number\t\tAge\t\tAddress\t\tCourses");
+		System.out.println("---------------------------------------------------------------------------------------------------------");
+		user.stream().sorted((a, b) -> {
+			int result = 0;
+			switch (ch) {
+			case 1:
+				result = a.getName().compareTo(b.getName());
+				break;
+			case 2:
+				result = a.getRollNumber() - b.getRollNumber();
+				break;
+			case 3:
+				result = a.getAge() - b.getAge();
+				break;
+			case 4:
+				result = a.getAddress().compareTo(b.getAddress());
+				break;
+			}
+			//If descending order is selected , invert the sign of result
+			if (order == 2) result = -result;
+			return result;
+		}).forEach(u -> u.display());
+		in.close();		
 	}
 	public static void deleteUser(Set<User> user) {
 		System.out.println("Enter roll number : ");
 		Scanner in = new Scanner(System.in);
```

Comment:
could have added input taking mechanism to a separate util file, and create functions there to takeIntInput, takeLineInput, etc. 
this would not require you to define `Scanner in = new Scanner(System.in);` again and again everywhere.

Reply:
Yea even i was thinking for the same